### ISSUE TITLE
(aws-cdk): Please consider picomatch or micromatch over minimatch

### ISSUE THREAD

**Issue by shellscape** [aws/aws-cdk#13825]
### The Question

In delving into the different ways I can use globs to signal which services to act upon with the CLI, I discovered that the aws-cdk is [making use of `minimatch`](https://github.com/aws/aws-cdk/blob/6f6e079569fcdb7e0631717fbe269e94f8f7b127/packages/aws-cdk/lib/api/cxapp/cloud-assembly.ts#L108). Astute choice, but not the most optimal IMHO. Both `picomatch` and `micromatch` have enjoyed more recent updates and share a wider feature set. Negating is still particularly troublesome with `minimatch`. Couple that with both alternative having the backing of most of the ecosystem's major toolsets these days, and I'm left very curious about the choice of using `minimatch`.

Would the project consider using one of the alternatives for a better end-user experience?

### Environment

  - **CDK CLI Version:** 1.95.1
  - **Module Version:** 1.95.1
  - **Node.js Version:** 14.5.3
  - **OS:** MacOS
  - **Language (Version):** TS

**Comment by peterwoodworth** [comment]
@rix0rrr thoughts on this?

**Comment by shellscape** [comment]
haha I guess he shared his thoughts with the removal of the assignment 😆

**Comment by peterwoodworth** [comment]
LOL he has way too many issues to look at I understand

I haven't looked into the alternatives you suggested much, can you explain why those would give the user a better experience?

> Would the project consider using one of the alternatives for a better end-user experience?

**Comment by rix0rrr** [comment]
Happy to consider improvements, as long as they are backwards compatible.

**Comment by shellscape** [comment]
I'd still like to circle back to this one

**Comment by peterwoodworth** [comment]
@shellscape the team won't have time to work on this soon, if you'd like to see this would you be willing to contribute and make a PR so we can discuss from there?

**Comment by shellscape** [comment]
Yes. Does this repo have a bot capable of timed reminders? I'm on vacation for a few weeks and won't get to this until back to reality

**Comment by peterwoodworth** [comment]
We don't, i'll write this down and ping you in a month. safe travels!

**Comment by TheRealAmazonKendra** [comment]
@shellscape Still interested in working on this?

**Comment by shellscape** [comment]
Thanks for the ping! definitely. it totally slipped off my radar.

**Comment by shellscape** [comment]
I've forked and have begun to work on this.

**Comment by madeline-k** [comment]
Thanks @shellscape! Looking forward to seeing your PR on this. If you are not able to work on it, we will probably close it within a few weeks. Please make sure any changes you are making are backwards compatible.

**Comment by shellscape** [comment]
I'm still here. Just keeping this fresh.

**Comment by shellscape** [comment]
I've run into cascading failing tests that I simply don't have the bandwidth to tackle at the moment. Sadly, after all this time, we'll have to let this one be closed by the bot or maintainers.
